src/util/amount.rs

```diff
@@ -605,6 +757,62 @@ impl ::core::iter::Sum for Amount {
     }
 }
 
+/// A helper/builder that displays amount with specified settings.
+///
+/// This provides richer interface than `fmt::Fromatter`:
+///
+/// * Ability to select denomination
+/// * Show or hide denomination
+/// * Dynamically-selected denomination - show in sats if less than 1 BTC. 
+///
+/// However this can still be combined with `fmt::Formatter` options to precisely control zeros,
+/// padding, alignment... The formatting works like floats from `core` but note that precision will
+/// be **never** lossy - that means no rounding.
+///
+/// See [`Amount::display_in`] and [`Amount::display_dynamic`] to construct this.
+#[derive(Debug, Clone)]
+pub struct Display {
+    /// Absolute value of satoshis to display (sign is below)
+    sats_abs: u64,
+    /// The sign
+    is_negative: bool,
+    /// How to display the value
+    style: DisplayStyle,
+}
+
+impl Display {
+    /// Makes subsequent calls to `Display::fmt` display denomination.
+    pub fn show_denomination(mut self) -> Self {
+        match &mut self.style {
+            DisplayStyle::FixedDenomination { show_denomination, .. } => *show_denomination = true,
+            // No-op because dynamic denomination is always shown
+            DisplayStyle::DynamicDenomination => (),
+        }
+        self
```

Comment:
```suggestion
	Self {
		style: match self.style {
		    DisplayStyle::FixedDenomination { denomination, .. } => DisplayStyle::FixedDenomination { show_denomination: true, denomination },
	            DisplayStyle::DynamicDenomination => DisplayStyle::DynamicDenomination,
		},
		..self
	}
```

Note: Untested for MSRV compatibility but this could be the non-mutable style of achieving the same thing (I think).

Reply:
Yeah no drama :)